app/services/katello/pulp3/repository/yum.rb

```diff
@@ -70,7 +70,35 @@ def partial_repo_path
           "/pulp/repos/#{repo.relative_path}/".sub('//', '/')
         end
 
-        def copy_units(source_repository, content_unit_hrefs, dependency_solving, dest_base_version = 0,
+        def multi_copy_units(repo_id_map, content_unit_hrefs, dependency_solving)
+          tasks = []
+
+          content_unit_hrefs.sort!
+          if content_unit_hrefs.any?
+            data = PulpRpmClient::Copy.new
+            data.dependency_solving = dependency_solving
+            data.config = []
+            repo_id_map.each do |source_repo_ids, dest_repo_id_map|
+              dest_repo = ::Katello::Repository.find(dest_repo_id_map[:dest_repo])
+              dest_repo_href = ::Katello::Pulp3::Repository::Yum.new(dest_repo, SmartProxy.pulp_master).repository_reference.repository_href
+              tasks << remove_all_content_from_repo(dest_repo_href)
```

Comment:
how would this work with incremental update?  Wouldn't this always clear out the content ?

Reply:
(and i'm very probably misunderstanding the flow here)

Reply:
The idea with incremental update is that the dest_base_version determines what content is should already be in the repo when copying.  `remove_all_content_from_repo` then would generate a new repo version which would not affect the incremental update.  Without the `remove_all_content_from_repo`, my exclude filtering does not work during content view publishing with dep solving.

Perhaps I could skip this content removal during incremental update by checking if `dest_repo_id_map[:base_version]` exists.

Reply:
```
unless dest_repo_id_map[:base_version]
  tasks << remove_all_content_from_repo(dest_repo_href)
end
```